torch/_refs/__init__.py

```diff
@@ -510,6 +512,38 @@ def _isnan(a: TensorLikeType) -> TensorLikeType:
 )
 
 
+@register_decomposition(torch.ops.aten.nan_to_num)
+@out_wrapper
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("a,"),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
+)
+def nan_to_num(
+    a: TensorLikeType,
+    *,
+    nan: Optional[NumberType] = 0.0,
+    posinf: Optional[NumberType] = None,
+    neginf: Optional[NumberType] = None,
+) -> TensorLikeType:
+    assert isinstance(a, TensorLike)
+
+    if posinf is None:
+        posinf = prims.maximum_value(a.dtype)
+
+    if neginf is None:
+        neginf = prims.minimum_value(a.dtype)
+
+    result = where(isnan(a), nan, a)
+
+    is_neg = signbit(a)
+    is_neginf = bitwise_and(isinf(a), is_neg)
+    result = where(is_neginf, neginf, result)
+
+    is_posinf = bitwise_and(isinf(a), bitwise_not(is_neg))
```

Comment:
Similar question about ordering -- could a NaN value be remapped to negative infinity, then remapped to positive infinity, then remapped by this? 

Do we have tests that this operation is consistent with eager's behavior when goofy cases like that come up?

Reply:
`nan_to_num` checks the original input before replacement, so values mapped to NaN, INF, -INF aren't remapped to another value.
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cpu/UnaryOpsKernel.cpp#L434-L440